include/boost/beast/core/async_base.hpp

```diff
@@ -229,7 +244,7 @@ class async_base
         Handler_&& handler,
         Executor1 const& ex1)
         : h_(std::forward<Handler_>(handler))
-        , wg1_(ex1)
+        , wg1_(detail::make_work_guard(ex1))
```

Comment:
Should this be called `make_tracked`?

Reply:
I don't know whether the net.ts or the execution nightmare is more likely to win out in future. I called it something I felt we would both woud understand.

Reply:
fair, and its in detail so users won't/shouldn't see it.